pkg/codegen/schema/schema.go

```diff
@@ -378,6 +380,64 @@ type Resource struct {
 	Methods []*Method
 }
 
+// The set of resource paths where ReplaceOnChanges is true.
+//
+// For example, if you have the following resource struct:
+//
+// Resource A {
+// Properties: {
+// 	 Resource B {
+// 	   Object D: {
+// 	     ReplaceOnChanges: true
+// 	     }
+// 	   Object F: {}
+//     }
+// 	 Object C {
+// 	   ReplaceOnChanges: true
+// 	   }
+//   }
+// }
+//
+// A.ReplaceOnChanges() == [[B, D], [C]]
+func (r *Resource) ReplaceOnChanges() (replaceOnChanges [][]*Property) {
+	for _, p := range r.Properties {
```

Comment:
@lblackstone, should we be looking at the resource's output properties (`Properties`), input properties (`InputProperties`), or both?

Reply:
We should check both inputs and outputs.

Reply:
The comment on `Properties` says it should be a superset of `InputProperties`, so I only check `Properties`. Is this wrong?

Reply:
> The comment on `Properties` says it should be a superset of `InputProperties`, so I only check `Properties`

That's generally the case, yes. I guess it depends on where we want the user to be able to specify these in the schema. And if we do end up looking at both, what happens if there's a conflict? @lblackstone, I'll defer to you on this.

Reply:
I think it should be fine to only check `Properties`.